xarray/tests/test_utils.py

```diff
@@ -297,7 +298,7 @@ def test_parse_dims_replace_none(dim: None | ellipsis) -> None:
         pytest.param(["x", 2], id="list_missing_all"),
     ],
 )
-def test_parse_dims_raises(dim: str | Iterable[Hashable]) -> None:
+def test_parse_dims_raises(dim):
```

Comment:
> the signature on the test matches the signature declared for the function being tested - quite pointless IMHO

I'm not completely sure what you mean by this. But without `check_untyped_defs`, having the `-> None` is the only way we test whether our annotations are correct (or am I wrong there?)

I think you should revert removing the annotations, at least the `-> None`

Reply:
What I was trying to say is that the signature in the unit test causes mypy to verify that the signature declared in the unit test, `dim: str | Iterable[Hashable]` is indeed compatible with the signature declared in the parse_dims function, `dim: Dims`. Which IMHO is pointless.

It would have been useful if it verified that the values with which `dim` is actually populated in the test are legal for the `Dims` type, but it does _not_ do that.

For example, in the future someone may short-sightedly change the definition of `Dims` from `str | Collection[Hashable]` to `str | Sequence[Hashable]`. One of the parameters in the test is 
```python
        pytest.param({"a", 1}, tuple({"a", 1}), id="non_sequence_collection"),
```
*Nothing will trip, with or without annotations in the test signature*, unless someone changes the actual implementation of `parse_dims` to crash if you pass a set to it.

Annotating the test signature would become useful if we rewrote it without parametrization:

```python
def test_parse_dims() -> None:
    all_dims = ("a", "b", 1, ("b", "c"))  # selection of different Hashables

    # non-sequence collection
    actual = utils.parse_dims({"a", 1}, all_dims, replace_none=False)
    assert actual == tuple({"a", 1})

    ... # repeat for all other use cases
```
which would be a valid choice, but it would fail on the first bad use case instead of moving on and it would be less immediately obvious what broke. You win some, you lose some.

**Would you like me to open a new PR where I rewrite the unit test without parametrization and with annotations?**

Reply:
https://github.com/pydata/xarray/pull/8618